src/Middleware/BaseAuthorizationMiddleware.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         1.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+
+namespace Authorization\Middleware;
+
+use Authorization\AuthorizationServiceInterface;
+use Authorization\AuthorizationServiceProviderInterface;
+use Authorization\Exception\AuthorizationRequiredException;
+use Authorization\Exception\Exception;
+use Authorization\IdentityInterface;
+use Authorization\Middleware\UnauthorizedHandler\HandlerFactory;
+use Authorization\Middleware\UnauthorizedHandler\HandlerInterface;
+use Cake\Core\InstanceConfigTrait;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use RuntimeException;
+
+/**
+ * Authorization Middleware.
+ *
+ * Injects the authorization service and decorated identity objects into the request object as attributes.
+ */
+abstract class BaseAuthorizationMiddleware implements MiddlewareInterface
```

Comment:
I'm not sure we *need* a base class. Having two copies of the `getHandler()` method is a better solution in my opinion as it uses fewer lines of code, and doesn't introduce options that aren't shared.

Reply:
I am generally under the impression that having a base class to share the common code is better than having the same exact method(s) copypasted over and over.